CRM/Contact/BAO/Contact.php

```diff
@@ -541,6 +541,10 @@ public static function ensureGreetingParamsAreSet(&$params) {
         //  If we are setting it to null then null out the display field.
         $params[$greetingIndex . '_display'] = 'null';
       }
+      if ((int) $params[$greetingParam] !== CRM_Core_PseudoConstant::getKey('CRM_Contact_BAO_Contact', $greetingParam, 'Customized')) {
```

Comment:
I'm curious about this strict operator. I'm not sure the `getKey` function can be relied upon to return strict integer types; it's an old function that afaik follow's PHP's default sloppy equivocation of numbers and strings.

Reply:
@colemanw I hate non-strict but I pulled it back in case it's unreliable